### PR TITLE
Start/stop P3A `MessageManager` depending on P3A enabled setting

### PR DESCRIPTION
**Pull request by DJAndries**
Resolves https://github.com/brave/brave-browser/issues/38712

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

Ensure measurements are sent to `collector.bsg.brave.com`, and requests are sent to `star-randsrv.bsg.brave.com` ONLY if P3A is enabled. Disable the `BraveP3ATypicalJSONDeprecation` and `BraveP3AOtherJSONDeprecation` features and ensure that requests are sent to `p3a-json.brave.com` ONLY if P3A is enabled.

### CODE REVIEW

File: components/p3a/p3a_service_unittest.cc [hunk 1]
```diff
@@ -82,6 +82,8 @@ class P3AServiceTest : public testing::Test {
     config_.p3a_creative_upload_url = GURL(kTestP3ACreativeHost);
   }
 
+  void TearDown() override { p3a_service_ = nullptr; }
```
Comment (iefremov) [inline components/p3a/p3a_service_unittest.cc:85]: why don't we `SetUpP3AService()` in `SetUp()`?

File: browser/ntp_background/ntp_p3a_helper_impl_unittest.cc [hunk 2]
```diff
@@ -71,6 +71,8 @@ class NTPP3AHelperImplTest : public testing::Test {
         &local_state_, p3a_service_.get(), &prefs_, true);
   }
 
+  void TearDown() override { p3a_service_ = nullptr; }
```
Comment (iefremov) [inline browser/ntp_background/ntp_p3a_helper_impl_unittest.cc:74]: why p3a service is not recreated in SetUp() ?

File: components/p3a/message_manager.cc [hunk 3]
```diff
@@ -122,6 +122,15 @@ void MessageManager::Init(
   }
 }
 
+void MessageManager::Stop() {
```
Comment (iefremov) [inline components/p3a/message_manager.cc:125]: we also check the preference in `StartScheduledUpload` and friends, maybe we can remove that check now or make it `CHECK` ?

**Review by iefremov — APPROVED** [review]

File: components/p3a/message_manager.cc (hunk 3 above)
Comment (DJAndries) [reply components/p3a/message_manager.cc:125]: `CHECK` sounds good

File: browser/ntp_background/ntp_p3a_helper_impl_unittest.cc (hunk 2 above)
Comment (DJAndries) [reply browser/ntp_background/ntp_p3a_helper_impl_unittest.cc:74]: it is

File: components/p3a/p3a_service_unittest.cc (hunk 1 above)
Comment (DJAndries) [reply components/p3a/p3a_service_unittest.cc:85]: In some tests (i.e. `UpdateLogsAndSendSlow`), we'd like to adjust the config before creating the service

**Pull request merged**